### PR TITLE
Andybranch

### PR DESCRIPTION
**Pull request by andygonzalez**
many changes to grumble

### CODE REVIEW

File: cmd/grumble/register.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	Password string   `xml:"password"`
+	Port     int      `xml:"port"`
+	Url      string   `xml:"url"`
+	Digest   string   `xml:"digest"`
+	Users    int      `xml:"users"`
+	Channels int      `xml:"channels"`
+	Location string   `xml:"location"`
+}
+
+const registerUrl = "https://mumble.hive.no/register.cgi"
```
Comment (colek42) [inline cmd/grumble/register.go]: NOOOOOOO!

File: cmd/grumble/server.go [hunk 2]
```diff
@@ -184,24 +179,59 @@ func (server *Server) RootChannel() *Channel {
...
+	parts := strings.Split(server.cfg.StringValue("SuperUserPassword"), "$")
+	if len(parts) != 3 {
+		return false
+	}
+
+	if len(parts[2]) == 0 {
+		return false
 	}
 
-	if password == supwd {
```
Comment (colek42) [inline cmd/grumble/server.go]: why are you removing this?  this simple check against the env var worked well

File: cmd/grumble/server.go [hunk 3]
```diff
@@ -46,22 +47,6 @@ const (
 	StateClientDead
 )
 
-var (
-	ActivePort = DefaultPort
-)
-
-func init() {
-	portStr := os.Getenv("GRUMBLE_PORT")
```
Comment (colek42) [inline cmd/grumble/server.go]: port needs to be set as an env var

File: cmd/grumble/register.go (hunk 1 above)
Comment (sthompson-Novetta) [inline cmd/grumble/register.go]: Andy,
We don't want this to register itself.  This isn't a public server. We want to exclude the registration code.

Unless you want us to also generate a common registration site for all our installations.  If that's the case, we need to discuss high-level architecture.

Either way, please don't have it talking to a public website.

File: cmd/grumble/server.go (hunk 3 above)
Comment (sthompson-Novetta) [inline cmd/grumble/server.go]: +1 to this.  We want to keep the default grumble port as a backup and testing, but we want to configure ports and such from env variables.

**Pull request merged**